sqlglot/dialects/spark.py

```diff
@@ -8,7 +8,13 @@
 
 def _create_sql(self, e):
     kind = e.args.get("kind")
-    temporary = e.args.get("temporary")
+    properties = e.args.get("properties")
+    if properties:
+        temporary = any(
+            [isinstance(prop, exp.TemporaryProperty) for prop in properties.expressions]
+        )
+    else:
+        temporary = None
 
     if kind.upper() == "TABLE" and temporary is True:
```

Comment:
if kind.upper() == "TABLE" and any(isinstance(prop, Temp) for prop in e.args.get("properties", []))

Reply:
What's the correct way to iterate over properties.expressions in this syntax?

`any(isinstance(prop, Temp) for prop in e.args.get("properties", []))`

Reply:
Alternatively, you could also do something like:
```python
properties = e.args.get("properties")
any(isinstance(prop, Temp) for prop in (properties.expressions if properties else []))
```

Reply:
if this is common make a method on the instance